glean-core/rlb/src/lib.rs

```diff
@@ -105,29 +112,139 @@ where
-    setup_state(AppState {
-        channel: cfg.channel,
-        client_info,
+/// See `glean_core::Glean::new` for more information.
+///
+/// # Arguments
+///
+/// * `cfg` - the `Configuration` options to initialize with.
+/// * `client_info` - the `ClientInfoMetrics` values used to set Glean
+///   core metrics.
+pub fn initialize(cfg: Configuration, client_info: ClientInfoMetrics) {
+    if was_initialize_called() {
+        log::error!("Glean should not be initialized multiple times");
+        return;
+    }
+
+    std::thread::spawn(move || {
+        let core_cfg = glean_core::Configuration {
+            upload_enabled: cfg.upload_enabled,
+            data_path: cfg.data_path.clone(),
+            application_id: cfg.application_id.clone(),
+            language_binding_name: LANGUAGE_BINDING_NAME.into(),
+            max_events: cfg.max_events,
+            delay_ping_lifetime_io: cfg.delay_ping_lifetime_io,
+        };
+
+        let glean = match Glean::new(core_cfg) {
+            Ok(glean) => glean,
+            // glean-core already takes care of logging errors: other bindings
+            // simply do early returns, as we're doing.
+            Err(_) => return,
+        };
+
+        // glean-core already takes care of logging errors: other bindings
+        // simply do early returns, as we're doing.
+        if glean_core::setup_glean(glean).is_err() {
+            return;
+        }
+
+        log::info!("Glean initialized");
+
+        // Now make this the global object available to others.
+        setup_state(RustBindingsState {
+            channel: cfg.channel,
+            client_info,
+        });
+
+        let upload_enabled = cfg.upload_enabled;
+
+        with_glean_mut(|glean| {
+            let state = global_state().lock().unwrap();
+
+            // Get the current value of the dirty flag so we know whether to
+            // send a dirty startup baseline ping below.  Immediately set it to
+            // `false` so that dirty startup pings won't be sent if Glean
+            // initialization does not complete successfully.
+            // TODO Bug 1672956 will decide where to set this flag again.
+            let dirty_flag = glean.is_dirty_flag_set();
+            glean.set_dirty_flag(false);
+
+            // Register builtin pings.
+            // Unfortunately we need to manually list them here to guarantee
+            // they are registered synchronously before we need them.
+            // We don't need to handle the deletion-request ping. It's never touched
+            // from the language implementation.
+            glean.register_ping_type(&glean_metrics::pings::baseline.ping_type);
+            glean.register_ping_type(&glean_metrics::pings::metrics.ping_type);
+            glean.register_ping_type(&glean_metrics::pings::events.ping_type);
+            /*
+                        // If any pings were registered before initializing, do so now.
```

Comment:
Better turn this into a proper `TODO` comment instead of copy-pasting kotlin code.

Reply:
Yup, sorry, I just added another commit that links to the related bugs.